Lib/fractions.py

```diff
@@ -168,9 +168,9 @@ def _round_to_figures(n, d, figures):
     # A '0' that's *not* followed by another digit is parsed as a minimum width
     # rather than a zeropad flag.
     (?P<zeropad>0(?=[0-9]))?
-    (?P<minimumwidth>0|[1-9][0-9]*)?
+    (?P<minimumwidth>[0-9]+)?
     (?P<thousands_sep>[,_])?
-    (?:\.(?P<precision>0|[1-9][0-9]*))?
+    (?:\.(?P<precision>[0-9]+))?
```

Comment:
For consistency I would also make the same change in `_GENERAL_FORMAT_SPECIFICATION_MATCHER`.

Reply:
This is more complicated. The first zero serves as the zeropad flag, but the zeropad flag is not supported in _GENERAL_FORMAT_SPECIFICATION_MATCHER. If we ignore leading zeroes, the zeropad flag will also be silently ignored. Currently an error is a signal that the zeropad flag is not supported.

Reply:
Okay, that's fair.